cg/meta/workflow/balsamic.py

```diff
@@ -232,6 +240,45 @@ def get_verified_normal_path(sample_data: dict) -> Optional[str]:
             return None
         return normal_paths[0]
 
+    def get_latest_raw_file_data(self, case_id: str, tags: list) -> Union[list, dict]:
+        """Retrieves the data of the latest file associated to a specific case ID and a list of tags"""
+
+        version = self.housekeeper_api.last_version(bundle=case_id)
+        raw_file = self.housekeeper_api.get_files(
+            bundle=case_id, version=version.id, tags=tags
+        ).first()
+
+        if not raw_file:
+            raise FileNotFoundError(
+                f"No file associated to {tags} was found in housekeeper for {case_id}"
+            )
+
+        with open(Path(raw_file.full_path), "r") as stream:
+            data = yaml.safe_load(stream)
+
+        return data
+
+    def get_latest_metadata(self, case_id: str) -> Union[BalsamicAnalysis, None]:
```

Comment:
why should this method return None sometimes, shouldn't it always exist metadata?

Reply:
@patrikgrenfeldt:

It actually mimics the MIP implementation:
https://github.com/Clinical-Genomics/cg/blob/2bebaae5d53dbf53d029800a1e695e32ba7eb4db/cg/meta/workflow/mip.py#L199

And it triggers a `DeliveryReportError` if its not able to retrieve the latest metadata during the report data collection:
https://github.com/Clinical-Genomics/cg/blob/2bebaae5d53dbf53d029800a1e695e32ba7eb4db/cg/meta/report/api.py#L131-L135

Is there a reason to keep this `None` return for MIP?